File: app/src/main/java/com/parabola/newtone/ui/fragment/settings/SettingFragment.java
package com.parabola.newtone.ui.fragment.settings;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import com.parabola.newtone.BuildConfig;
import com.parabola.newtone.MainApplication;
import com.parabola.newtone.R;
import com.parabola.newtone.databinding.FragmentSettingBinding;
import com.parabola.newtone.di.app.AppComponent;
import com.parabola.newtone.mvp.presenter.SettingPresenter;
import com.parabola.newtone.mvp.view.SettingView;
import com.parabola.newtone.ui.base.BaseSwipeToBackFragment;

import java.util.Locale;

import moxy.presenter.InjectPresenter;
import moxy.presenter.ProvidePresenter;

public final class SettingFragment extends BaseSwipeToBackFragment
        implements SettingView {
    private static final String LOG_TAG = SettingFragment.class.getSimpleName();

    private static final String VERSION_INFO;

    static {
        VERSION_INFO = String.format(Locale.getDefault(), "v.%s. build-%d", BuildConfig.VERSION_NAME, BuildConfig.VERSION_CODE);
    }

    private FragmentSettingBinding binding;


    @InjectPresenter SettingPresenter presenter;


    public SettingFragment() {
        // Required empty public constructor
    }


    @NonNull
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = super.onCreateView(inflater, container, savedInstanceState);
        binding = FragmentSettingBinding.inflate(inflater, container, false);
        getRootBinding().container.addView(binding.getRoot());


        getRootBinding().main.setText(R.string.setting_title);

        getRootBinding().additionalInfo.setVisibility(View.GONE);
        getRootBinding().otherInfo.setVisibility(View.GONE);
        binding.appInfoVersion.setText(VERSION_INFO);

        binding.colorThemeBar.setOnClickListener(v -> presenter.onClickColorThemeSettings());
        binding.notificationColorBar.setOnClickListener(v -> presenter.onClickNotificationColorSetting());
        binding.notificationArtworkShowBar.setOnClickListener(v -> presenter.onClickNotificationArtworkShowSetting());
        binding.showItemDividerBar.setOnClickListener(v -> presenter.onClickShowItemDivider());
        binding.excludedFoldersBar.setOnClickListener(v -> presenter.onClickExcludedFolders());
        binding.trackItemViewBar.setOnClickListener(v -> presenter.onClickTrackItemViewSettings());
        binding.albumItemViewBar.setOnClickListener(v -> presenter.onClickAlbumItemViewSettings());
        binding.artistItemViewBar.setOnClickListener(v -> presenter.onClickArtistItemViewSettings());
        binding.privacyPolicyBar.setOnClickListener(v -> presenter.onClickPrivacyPolicy());
        binding.contactDevelopersBar.setOnClickListener(v -> presenter.onClickContactDevelopers());
        binding.appInfoBar.setOnClickListener(v -> presenter.onClickAppInfoBar());

        return root;
    }

    @Override
    protected void onClickBackButton() {
        presenter.onClickBack();
    }


    @Override
    public void setNotificationArtworkSwitchChecked(boolean isChecked) {
        binding.notificationArtworkShowSwitch.setChecked(isChecked);
        setNotificationColorBarEnabling(isChecked);
    }

    private void setNotificationColorBarEnabling(boolean checked) {
        binding.notificationColorBar.setEnabled(checked);
    }

    @Override
    public void setNotificationColorSwitchChecked(boolean isChecked) {
        binding.notificationColorSwitch.setChecked(isChecked);
    }

    @Override
    public void setShowListItemDividerSwitchChecked(boolean isChecked) {
        binding.showItemDividerSwitch.setChecked(isChecked);
    }

    @ProvidePresenter
    public SettingPresenter providePresenter() {
        AppComponent appComponent = ((MainApplication) requireActivity().getApplication()).getAppComponent();
        return new SettingPresenter(appComponent);
    }

}

File: app/src/main/java/com/parabola/newtone/presentation/settings/SettingFragment.kt
package com.parabola.newtone.presentation.settings

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.parabola.newtone.BuildConfig
import com.parabola.newtone.MainApplication
import com.parabola.newtone.R
import com.parabola.newtone.databinding.FragmentSettingBinding
import com.parabola.newtone.presentation.base.BaseSwipeToBackFragment
import moxy.presenter.InjectPresenter
import moxy.presenter.ProvidePresenter
import java.util.Locale

class SettingFragment : BaseSwipeToBackFragment(), SettingView {

    companion object {
        private var VERSION_INFO: String = String.format(
            Locale.getDefault(),
            "v.%s. build-%d",
            BuildConfig.VERSION_NAME,
            BuildConfig.VERSION_CODE
        )
    }

    private var _binding: FragmentSettingBinding? = null
    private val binding get() = _binding!!

    @InjectPresenter
    lateinit var presenter: SettingPresenter

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val root = super.onCreateView(inflater, container, savedInstanceState)
        _binding = FragmentSettingBinding.inflate(inflater, container, false)

        rootBinding.container.addView(binding.root)

        rootBinding.main.setText(R.string.setting_title)
        rootBinding.additionalInfo.visibility = View.GONE
        rootBinding.otherInfo.visibility = View.GONE

        binding.apply {
            appInfoVersion.text = VERSION_INFO

            colorThemeBar.setOnClickListener { presenter.onClickColorThemeSettings() }
            notificationColorBar.setOnClickListener { presenter.onClickNotificationColorSetting() }
            notificationArtworkShowBar.setOnClickListener { presenter.onClickNotificationArtworkShowSetting() }
            showItemDividerBar.setOnClickListener { presenter.onClickShowItemDivider() }
            excludedFoldersBar.setOnClickListener { presenter.onClickExcludedFolders() }
            trackItemViewBar.setOnClickListener { presenter.onClickTrackItemViewSettings() }
            albumItemViewBar.setOnClickListener { presenter.onClickAlbumItemViewSettings() }
            artistItemViewBar.setOnClickListener { presenter.onClickArtistItemViewSettings() }
            privacyPolicyBar.setOnClickListener { presenter.onClickPrivacyPolicy() }
            contactDevelopersBar.setOnClickListener { presenter.onClickContactDevelopers() }
            appInfoBar.setOnClickListener { presenter.onClickAppInfoBar() }
        }

        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }


    override fun onClickBackButton() {
        presenter.onClickBack()
    }

    override fun setNotificationArtworkSwitchChecked(isChecked: Boolean) {
        binding.notificationArtworkShowSwitch.isChecked = isChecked
        setNotificationColorBarEnabling(isChecked)
    }

    private fun setNotificationColorBarEnabling(checked: Boolean) {
        binding.notificationColorBar.isEnabled = checked
    }

    override fun setNotificationColorSwitchChecked(isChecked: Boolean) {
        binding.notificationColorSwitch.isChecked = isChecked
    }

    override fun setShowListItemDividerSwitchChecked(isChecked: Boolean) {
        binding.showItemDividerSwitch.isChecked = isChecked
    }

    @ProvidePresenter
    fun providePresenter(): SettingPresenter {
        val appComponent = (requireActivity().application as MainApplication).appComponent
        return SettingPresenter(appComponent)
    }
}
